include/swift/AST/Requirement.h

```diff
@@ -124,6 +124,55 @@ struct StructuralRequirement {
   bool inferred = false;
 };
 
+/// Represents an invalid requirement, such as `T: Int`.
```

Comment:
Do you think this will be used outside of lib/AST/RequirementMachine? If not, we could put this in lib/AST/RequirementMachine/Diagnostics.h or something.